src/NServiceBus.Core/obsoletes.cs

```diff
@@ -2254,6 +2255,24 @@ public void ApplyTo(Type componentType, IComponentConfig config)
         {
             throw new NotImplementedException();
         }
+
+        [ObsoleteEx(
+          RemoveInVersion = "7.0",
+          TreatAsErrorFromVersion = "6.0",
+          Message = "Set(string key, object value)")]
```

Comment:
I think when you do it like you did you have to use ReplacementTypeOrMember

But I don't think this is enough. I would appreciate a more thorough explanation like

Setting properties on an object is no longer supported. Set the value in the settings dictionary by using Set(string key, object value) and pass the settings holder to the object needing the settings. Extract the settings and set the state accordingly.

Sorry just typed this out of the void but something similar.